GoogleForCreators/web-stories-wp
# Media Hotlinking: Add initial version

## Context
Adds the possibility to add media by inserting a link.


## Summary



## Relevant Technical Choices



## To-do

- [ ] Some refactoring



## User-facing changes
<img width="565" alt="Screenshot 2021-08-11 at 18 38 58" src="https://user-images.githubusercontent.com/3294597/129059947-d6e9a0b1-7cb4-42cd-8b22-405bdec57ba4.png">
<img width="359" alt="Screenshot 2021-08-11 at 18 38 53" src="https://user-images.githubusercontent.com/3294597/129059957-e04faa37-2e72-4207-9e40-f6c4776e753b.png">



## Testing Instructions
Feature flag for hotlinking must be enabled first.
1. Click on the new "Insert by link" button.
2. Try inserting images/videos via link.
3. verify that invalid links show an error.
4. Verify that valid links create a new media element correctly.
5. If the user has permissions to upload media, verify that for a video inserted by a link, a poster image is created, too.
6. Verify that everything looks as before when disabling the hotlinking feature.

Note: Not all sites allow linking externally like this, in that case, an error starting with "Media failed to load" will be displayed under the input.




### QA






This PR can be tested by following these steps:

1.

## Checklist




---



Fixes #471
